includes/wccom-site/class-wc-wccom-site.php

```diff
@@ -124,11 +124,17 @@ protected static function get_authorization_header() {
 	 * @return bool
 	 */
 	protected static function is_request_to_wccom_site_rest_api() {
-		$request_uri = add_query_arg( array() );
-		$rest_prefix = trailingslashit( rest_get_url_prefix() );
-		$request_uri = esc_url_raw( wp_unslash( $request_uri ) );
+		$route = '/';
+		$rest_prefix = '';
```

Comment:
It seems like `$route` value is overwritten anyway, so probably don't need it here at all. 

And `$rest_prefix = '';` could probably be placed within the `if ( isset( $_REQUEST['rest_route'] ) )` condition, as it is also overwritten if this condition is not met.

Reply:
@jbutkus the problemwith the current solution is that is generating a whole new URL from that query string, also doesn't create the correct URL.

`$route` is equal to `http://wccom-site/v1/installer`, and `$rest_prefix` equal to `wccom-site/`, so no need to `esc_url_raw` in this case, also it's just to compare, so no need to sanitize like that in this case.